politeiad/backendv2/tstorebe/tstore/client.go

```diff
@@ -0,0 +1,448 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package tstore
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	backend "github.com/decred/politeia/politeiad/backendv2"
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/plugins"
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/store"
+	"github.com/google/trillian"
+	"google.golang.org/grpc/codes"
+)
+
+// BlobSave saves a BlobEntry to the tstore instance. The BlobEntry will be
+// encrypted prior to being written to disk if the record is unvetted. The
+// digest of the data, i.e. BlobEntry.Digest, can be thought of as the blob ID
+// and can be used to get/del the blob from tstore.
+//
+// This function satisfies the plugins TstoreClient interface.
+func (t *Tstore) BlobSave(treeID int64, be store.BlobEntry) error {
+	log.Tracef("BlobSave: %v", treeID)
+
+	// Verify tree exists
+	if !t.TreeExists(treeID) {
+		return backend.ErrRecordNotFound
+	}
+
+	// Verify tree is not frozen
+	leaves, err := t.tlog.LeavesAll(treeID)
```

Comment:
Its kinda expensive (100ms of latency for every 10k leaves) but there is not a better way to do this.

Reply:
Can we stuff this in the db?

Reply:
We could if we need to, but I don't think we need to.  This will only be expensive for plugin writes on trees with tens of thousands of leaves.  A tree having that many leaves is going to be a rare occurance.  It currently only happens at the end of ticket votes, which the record is locked against further writes in that scenrio anyway.

Stuffing it in the database opens up the possibility of it becoming incoherent due to a bug, allowing writes to a record that is suppose to be locked.